modules/basics/src/main/java/com/opengamma/basics/date/ImmutableHolidayCalendar.java

```diff
@@ -165,12 +202,154 @@ public LocalDateRange getRange() {
   @Override
   public boolean isHoliday(LocalDate date) {
     ArgChecker.notNull(date, "date");
-    if (range.contains(date) == false) {
+    try {
+      // find data for month
+      int index = (date.getYear() - startYear) * 12 + date.getMonthValue() - 1;
+      // check if bit is 1 at zero-based day-of-month
+      return (lookup[index] & (1 << (date.getDayOfMonth() - 1))) > 0;
```

Comment:
Any value of (tediously) statically creating the bit masks for each day up front i.e. 
```
int[] masks = new int[31];
for (int i = 0; i < 31; i++) {
  masks[i] = (1 << i);
}
```
Then in the line here, you can just do:
```
return (lookup[index] & masks[date.getDayOfMonth() - 1)]) > 0;
```
(As before re: indexxing of day in month)